### PR TITLE
Send post move survey email

### PR DESCRIPTION
**Pull request by ralren**
## Description
This deals with making the command line task to send a survey email fifteen days after a move's payment request has been reviewed. Running the command multiple times should not send multiple emails. Another PR will deal with setting this up to be run in AWS. Look here for the contents of the email: https://docs.google.com/document/d/1VLNbT5_cX6P-MX7TgFAXWajp71YG99KZvUZrvQM2qBY/edit#heading=h.cdt9fip3wk8x

## Setup

```sh
make server_run
make office_client_run
go run ./cmd/send_post_move_survey_email
```

Also test the integration in local dev with SES by:

1) `make db_dev_e2e_populate` to add sample moves
2) In `personally_procured_moves` choose a row and set `review_date` to 15 days minus today.
3) For the service member associated with the move above update `service_members.personal_email` to your email.
4) run `go run ./cmd/send_post_move_survey_email --email-backend=ses --aws-ses-domain=devlocal.dp3.us --aws-ses-region=us-west-2` you should receive an email.
5) Run the above command again. No email should be sent this time.

## Code Review Verification Steps
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168035083) for this change

### CODE REVIEW

**Comment by mkrump** [comment]
@chrisgilmerproj I think we addressed all of your comments if you want to take another look (check out the updated setup section). There is not a separate notifications table that tracks the email sent, along with the id returned by SES, which we could use to get additional info from SES if we want to.

**Comment by chrisgilmerproj** [comment]
Looks like this is almost done. There's an issue with the template not being in the right location in the docker file: 

```
make tasks_send_post_move_surveyBuild the docker scheduled tasks container...
docker build -f Dockerfile.tasks_local --tag tasks:latest .
Sending build context to Docker daemon    501MB
Step 1/4 : FROM gcr.io/distroless/base:latest
 ---> 7f04a8d24717
Step 2/4 : COPY bin_linux/save-fuel-price-data /bin/save-fuel-price-data
 ---> Using cache
 ---> ed4b6f5f9a05
Step 3/4 : COPY bin_linux/send-post-move-survey-email /bin/send-post-move-survey-email
 ---> Using cache
 ---> dcad997383dc
Step 4/4 : WORKDIR /bin
 ---> Using cache
 ---> 8467369b9473
Successfully built 8467369b9473
Successfully tagged tasks:latest
sending post move survey with docker command...
DB_NAME=dev_db DB_DOCKER_CONTAINER=milmove-db-dev scripts/wait-for-db-docker
> docker exec milmove-db-dev psql postgres://postgres:mysecretpassword@localhost:5432/dev_db -c 'select 1;' > /dev/null 2>&1
Checking for database connectivity..success!
docker run \
                -t \
                -e DB_HOST="database" \
                -e DB_NAME \
                -e DB_PORT \
                -e DB_USER \
                -e DB_PASSWORD \
                --link="milmove-db-dev:database" \
                --rm \
                tasks:latest \
                send-post-move-survey-email
panic: asset: Asset(pkg/notifications/templates/move_reviewed_template.txt): Asset pkg/notifications/templates/move_reviewed_template.txt not found

goroutine 1 [running]:
github.com/transcom/mymove/pkg/assets.MustAsset(0xfe50f5, 0x36, 0xc000559f58, 0xda9e5d, 0xc000559f68)
        /Users/cgilmer/Projects/transcom/mymove/pkg/assets/assets.go:160 +0x111
github.com/transcom/mymove/pkg/notifications.init.ializers()
        /Users/cgilmer/Projects/transcom/mymove/pkg/notifications/move_reviewed.go:24 +0x3a
make: *** [tasks_send_post_move_survey] Error 2
```

My thinking is that the command ought to let you pass through the location of the template file like `--template /path/to/template/move_reviewed_template.txt` so that we can set it in an environment variable. Currently docker is looking for that template at the pkg/notifications/templates/move_reviewed_template.txt location which is specific to our git source tree and would not make sense inside our dockerfile.

**Comment by mkrump** [comment]
```bash
make tasks_send_post_move_survey
Build the docker scheduled tasks container...
docker build -f Dockerfile.tasks_local --tag tasks:latest .
Sending build context to Docker daemon  298.8MB
Step 1/4 : FROM gcr.io/distroless/base:latest
 ---> 7f04a8d24717
Step 2/4 : COPY bin_linux/save-fuel-price-data /bin/save-fuel-price-data
 ---> Using cache
 ---> 9d35e67c00d6
Step 3/4 : COPY bin_linux/send-post-move-survey-email /bin/send-post-move-survey-email
 ---> Using cache
 ---> d75a432e2ad7
Step 4/4 : WORKDIR /bin
 ---> Using cache
 ---> 2e2806984dc5
Successfully built 2e2806984dc5
Successfully tagged tasks:latest
sending post move survey with docker command...
DB_NAME=dev_db DB_DOCKER_CONTAINER=milmove-db-dev scripts/wait-for-db-docker
> docker exec milmove-db-dev psql postgres://postgres:mysecretpassword@localhost:5432/dev_db -c 'select 1;' > /dev/null 2>&1
Checking for database connectivity..success!
docker run \
		-t \
		-e DB_HOST="database" \
		-e DB_NAME \
		-e DB_PORT \
		-e DB_USER \
		-e DB_PASSWORD \
		--link="milmove-db-dev:database" \
		--rm \
		tasks:latest \
		send-post-move-survey-email
2019-09-03T18:33:58.385Z	DEBUG	send_post_move_survey_email/main.go:30	checking config
2019-09-03T18:33:58.385Z	INFO	cli/dbconn.go:272	Connecting to the database	{"url": "postgres://postgres:*****@database:5432/dev_db?sslmode=disable", "db-ssl-root-cert": ""}
2019-09-03T18:33:58.389Z	INFO	notifications/notification_sender.go:79	Using local email backend	{"domain": "milmovelocal"}
2019-09-03T18:33:58.396Z	INFO	notifications/move_reviewed.go:88	no emails to be sent for	{"date": "2019-08-19 18:33:58.3892321 +0000 UTC"}
```

works on my machine :) just kidding, but seriously. my understanding of how `assets` pkg works is that it converts all of these forms to binary and when it comes time to load a file it looks in `assets.go` first. Only if it doesn't find it there does it try to look to the path. `assets.go` should get generated with the build. if you open that file you should see binary version of the templates.

**Comment by rdhariwal** [comment]
image
 Ran successfully for me. I ran `make clean' and then then after I had migrated the db `make tasks_send_post_move_survey` and it ran without any probs!

File: Makefile [hunk 1]
```diff
@@ -253,11 +253,17 @@ bin/save-fuel-price-data: .server_generate.stamp
 bin_linux/save-fuel-price-data: .server_generate_linux.stamp
 	GOOS=linux GOARCH=amd64 go build -ldflags "$(LDFLAGS)" -o bin_linux/save-fuel-price-data ./cmd/save_fuel_price_data
 
+bin/send-post-move-survey-email: .server_generate.stamp
```
Comment (chrisgilmerproj) [inline Makefile]: Both of these need to be changed to this:

```suggestion
bin/send-post-move-survey-email: .server_generate.stamp pkg/assets/assets.go
```

File: Makefile [hunk 2]
```diff
@@ -253,11 +253,17 @@ bin/save-fuel-price-data: .server_generate.stamp
...
+	go build -ldflags "$(LDFLAGS)" -o bin/send-post-move-survey-email ./cmd/send_post_move_survey_email
+
+bin_linux/send-post-move-survey-email: .server_generate_linux.stamp
```
Comment (chrisgilmerproj) [inline Makefile]:
```suggestion
bin_linux/send-post-move-survey-email: .server_generate_linux.stamp pkg/assets/assets.go
```

File: Makefile (hunk 1 above)
Comment (mkrump) [reply Makefile]: added dependency to both targets.

**Comment by mkrump** [comment]
@chrisgilmerproj most recent series of commits b62b261 8154296 6a58e00 a262275 fc4d6c4 cover changes you asked for.

**Pull request merged**